### PR TITLE
✨ feat: Add offline cache for lookup ens (1 week)

### PR DESCRIPTION
**Pull request by insulineru**

### CODE REVIEW

File: src/utils/evm.ts [hunk 1]
```diff
@@ -29,5 +36,6 @@ export const isEvmContract = async (chainId: number, address: string) => {
   const { rpc_url: rpc } = getNetworkById(chainId)
   const provider = new ethers.providers.JsonRpcProvider(rpc)
   const addressCode = await provider.getCode(address)
+  console.log('[isEvmContract]')
```
Comment (iower) [inline src/utils/evm.ts:39]:
```suggestion
```

**Review by iower — APPROVED** [review]
Looks OK

File: src/constants.ts [hunk 2]
```diff
@@ -138,3 +139,5 @@ export const ERC20_GAS_LIMIT = 300000
 export const LOCAL_STORAGE_WALLETS_KEY = 'web3-wallets-data'
 
 export const LOCAL_STORAGE_WALLETS_ADDRESSES = 'web3-wallets-addresses'
+
+export const EVM_PROVIDER = new ethers.providers.JsonRpcProvider('https://rpc.ankr.com/eth')
```
Comment (nrjshka) [inline src/constants.ts]: `https://rpc.ankr.com/eth` to const

Comment (nrjshka) [reply src/constants.ts]: and use const to initialize `EVM_PROVIDER`

File: src/context/QueryProvider.tsx [hunk 3]
```diff
@@ -1,16 +1,32 @@
...
 
-const queryClient = /* #__PURE__ */ new QueryClient()
+const queryClient = /* #__PURE__ */ new QueryClient({
+  defaultOptions: {
+    queries: {
+      cacheTime: 1000 * 60 * 60 * 24 * 7, // 1 week
+      staleTime: 1000 * 60 * 60 * 2 // 2 hours
+    }
+  },
+})
```
Comment (nrjshka) [inline src/context/QueryProvider.tsx:17]: We use `react-query` in `useBalances` as well. With such settings it might be affected. Have you checked it?

Comment (insulineru) [reply src/context/QueryProvider.tsx:17]: All balances queries has own `refetchInterval`

**Pull request merged**